src/lib/index.js

```diff
@@ -1159,3 +1159,31 @@ lib.fillText = function(calcPt, trace, contOut) {
 lib.isValidTextValue = function(v) {
     return v || v === 0;
 };
+
+lib.formatPercent = function(ratio, n) {
+    n = n || 0;
+    var str = (Math.round(100 * ratio * Math.pow(10, n)) * Math.pow(0.1, n)).toFixed(n) + '%';
+    for(var i = 0; i < n; i++) {
+        if(str.indexOf('.') !== -1) {
+            str = str.replace('0%', '%');
+            str = str.replace('.%', '%');
+        }
+    }
+    return str;
+};
+
+lib.convexPolygonArea = function(points) {
```

Comment:
Looks like you're only using this function in `funnelarea_test.js`. Please move it to `funnelarea_test.js`, we don't want to bundle things we don't use in `src/`.

Reply:
Done in https://github.com/plotly/plotly.js/pull/3876/commits/fa5b27332b474e86c1236e9af4c1ccdcbb7312ce. Also more tests were added for calculated areas in different `scalegroups`.